packages/graphiql/src/components/DocExplorer.js

```diff
@@ -54,6 +57,29 @@ export class DocExplorer extends React.Component {
     );
   }
 
+  componentDidUpdate(prevProps) {
+    if (this.props.schema && 
+      (this.props.schema !== prevProps.schema) && 
+      this.props.initialTypeName) {
+      // The schema has changed, and we should render an initial type.
```

Comment:
I'm not sure that we should do this on componentDidUpdate unless there was no previous schema. My reasoning is that it will clash with the desired "hot load" functionality, where the navStack is maintained over schema replacements, because it will keep pushing the type/field onto the nav stack (assuming it's not already at the top). I think maybe only do it if the nav stack is empty (one entry)?